pandas/tests/dtypes/test_inference.py

```diff
@@ -834,6 +834,31 @@ def test_infer_dtype_datetime(self):
         arr = np.array([np.nan, "2011-01-01", pd.Timestamp("2011-01-02")])
         assert lib.infer_dtype(arr, skipna=True) == "mixed"
 
+    @pytest.mark.parametrize("na", [pd.NaT, np.nan, np.datetime64("NaT"), None, pd.NA])
+    def test_infer_dtypes_datetimetz_with_na(self, na):
+
+        arr = np.array([na, datetime(2011, 1, 1, tzinfo=pytz.utc)])
+        assert lib.infer_dtype(arr, skipna=True) == "datetimetz"
+
+        arr = np.array([na, pd.Timestamp("2011-01-02", tzinfo=pytz.utc)])
+        assert lib.infer_dtype(arr, skipna=True) == "datetimetz"
+
+    def test_infer_dtypes_datetimetz_diff_tz(self):
+
+        # datetimes with different timezones
+        arr = [
+            datetime(2011, 1, 2, tzinfo=pytz.utc),
+            datetime(2011, 1, 2, tzinfo=pytz.timezone("US/Eastern")),
+        ]
+        assert lib.infer_dtype(arr, skipna=True) == "datetimetz"
+
+        # pd.Timestamps with different timezones
+        arr = [
+            pd.Timestamp("2011-01-02", tz="UTC"),
```

Comment:
you can use `pytest.mark.parameterize` over `[datetime, Timestamp]` and construct these like the `datetime` form you have above

Reply:
Yep... makes sense! Don't know why I did not think about it! I will make a new commit after I do some tests related to the request of @jreback.